packages/synapse-client/src/util/fetchWithExponentialTimeout.test.ts

```diff
@@ -0,0 +1,128 @@
+import { NETWORK_UNAVAILABLE_MESSAGE } from './Constants'
+import { fetchWithExponentialTimeout } from './fetchWithExponentialTimeout'
+import { SynapseClientError } from './SynapseClientError'
+
+describe('fetchWithExponentialTimeout', () => {
```

Comment:
One of these tests was written in the tests for `synapse-react-client/src/synapse-client/SynapseClient`, and I was having trouble getting Vite to properly mock `fetch` in that test for some reason.

Anyway, it was really testing `fetchWithExponentialTimeout`, so I moved the test here and added a few more.